src/libraries/System.Threading.Tasks.Parallel/src/System/Threading/Tasks/Parallel.ForEachAsync.cs

```diff
@@ -481,17 +481,40 @@ public void Complete()
                 else
                 {
                     // Fault with all of the received exceptions, but filter out those due to inner cancellation,
-                    // as they're effectively an implementation detail and stem from the original exception.
+                    // as they're effectively an implementation detail and stem from the original exception. However,
```

Comment:
What happens if instead you don't filter any exception?

Reply:
Some iteration of the loop throws an exception,  say an IOException, which causes all other in-flight operations to be canceled, so they throw OperationCanceledException. The idea is that those N-1 exceptions are all symptoms rather than the actual error,  and so should be filtered out to reduce the noise.

Reply:
We never know whether `internalCt.IsCancellationRequested` is checked by the developer when they throw `OperationCanceledException(internalCt)`, so we can't tell it's an actual error or noise.

Reply:
> We never know whether internalCt.IsCancellationRequested is checked by the developer when they throw OperationCanceledException(internalCt), so we can't tell it's an actual error or noise.

Yes, but such usage doesn't make a lot of sense.